### DISCUSSION TITLE
[Feature Request]: category.yml文件能不能与媒体目录结合

### DISCUSSION THREAD

**Discussion by WhaleHu** [jxxghp/MoviePilot#3349] [讨论]
### 当前程序版本

1.9.7

### 运行环境

Docker

### 功能改进类型

主程序

### 功能改进

category.yml文件能不能与媒体目录结合，一级名字为媒体目录别名，或者直接在媒体目录下配置自动分类规则？
这样个人定制容易了

**Comment by zoffyultraman** [comment]
目前不是更容易定制吗，category.yml决定了在目录设置中有哪些1级和2级分类可以选，设置可以把不同的2级分类指定到不同的文件夹。

**Comment by WhaleHu** [comment]
> 目前不是更容易定制吗，category.yml决定了在目录设置中有哪些1级和2级分类可以选，设置可以把不同的2级分类指定到不同的文件夹。

@zoffyultraman 举一个例子
* 电视剧
    * 大陆
    * 日韩
    * 美剧
* 综艺
    * 大陆
    * 其他
* 番剧 
    * 大陆
    * 日本
    * 其他
 这样的目录剧集和番剧差异巨大，没想好怎么实现
如果category.yml是一级和目录一级同名很容易实现了

**Comment by zoffyultraman** [comment]
存储结构，和2级分类结构请分开理解。二级分类是程序判断影片属于什么类型，你完全可以电视剧下分这些栏目，媒体目录直接指定文件夹。这么说可以理解吗，你这个需求直接目前就可以实现。给你举个例子。
tv:
  国产剧:
    origin_country: 'CN,TW,HK'
  欧美剧:
    origin_country: 'US,FR,GB,DE,ES,IT,NL,PT,RU,UK'
  日韩剧:
    origin_country: 'JP,KP,KR,TH,IN,SG'
  大陆综艺:
    genre_ids: '10764,10767'
    origin_country: 'CN'
  其他综艺:
    genre_ids: '10764,10767'

媒体目录假设为/media/links
媒体目录直接设置对应的即可
你可以/media/links/tv/综艺/大陆 综艺直接设置电视剧-大陆综艺
其他也雷同，emby/jellyfin添加的时候直接添加即可
![image](https://github.com/jxxghp/MoviePilot/assets/18696463/4c827dbf-7f61-4998-91db-3f390ed5c7f8)
此时识别到该情况，会自动处理到对应媒体目录
1级只有电影和电视剧是因为tmdb严格意义上只有这两大分类，剩下的子类全部属于这俩的其中之一

**Comment by WhaleHu** [comment]
@zoffyultraman 你说的根本不是一回事，我目的就是区分tv和日漫和纪录片

**Comment by zoffyultraman** [comment]
> @zoffyultraman 你说的根本不是一回事，我目的就是区分tv和日漫和纪录片

你只需要在媒体目录中随便指定你想要的目录即可。在依赖tmdb的情况下，日漫，纪录片就是隶属于tv，这个你没办法改变，你可以直接把日漫的目录指定到/media/links/日漫 可以抛弃掉目录中的tv这一层，虽然二级分类还隶属于tv，但是实际上目录和这个没有任何关系

**Comment by zoffyultraman** [comment]
![image](https://github.com/jxxghp/MoviePilot/assets/18696463/4ca3293e-a584-4997-a7af-7c93d1dc2b96)
通过二级分类过滤完后，媒体目录直接对应目录入库即可，现阶段这些不可能从tv里剥离了。
最主要的是tmdb只有这两个分类，剩下的都是隶属于其中之一子类，你可以在文件夹上把他们从tv分离开，但二级分类不可能把他们从tv里面剥离了。

**Comment by WhaleHu** [comment]
> ![image](https://private-user-images.githubusercontent.com/18696463/342068904-4ca3293e-a584-4997-a7af-7c93d1dc2b96.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkxNTkxMTYsIm5iZiI6MTcxOTE1ODgxNiwicGF0aCI6Ii8xODY5NjQ2My8zNDIwNjg5MDQtNGNhMzI5M2UtYTU4NC00OTk3LWE3YWYtN2M5M2QxZGMyYjk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIzVDE2MDY1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5MWFlYjA1Y2QxZDc4ODU0NzE3OTYxNDQ5ZmU3Mzc2MzI1OWIxYjJhMWJlYzc5YjQxMzU2NjA2OWYwZmQ5MjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xzanyL9ZB7vDmDk8dWstF1efuYHI8EylfNhCzd--GTg) 通过二级分类过滤完后，媒体目录直接对应目录入库即可，现阶段这些不可能从tv里剥离了。 最主要的是tmdb只有这两个分类，剩下的都是隶属于其中之一子类，你可以在文件夹上把他们从tv分离开，但二级分类不可能把他们从tv里面剥离了。

现在这个模式就要是不同目录就类似多if的嵌套，需要反复调试，关键是tmdb现在会有一小部分冷门的东西，你条件写的太那啥就会不准，所以目录同步我就大类分开了，纪录片下载时直接手动放在专门的。这种番剧更新中和美剧更新中类似就要好多条实现可能就会出现匹配不上的。

tmdb分类可以从媒体库类型读取呀。然后直接匹配二级，相当于顺序换一下，这个文件同时规定了二级分类和二级分类对应的目录，执行时候生成一个每个文件夹自己category.yml。

说白了就是每个媒体目录都写一个自己匹配规则也行。里面还可以tv： 类似通用.yaml a.yaml 这种，a目录优先用同名的

**Comment by zoffyultraman** [comment]
目前是有优先级匹配的，你只需要把番剧的靠前放就可以了。
我现在已经有点懵逼了
“tmdb分类可以从媒体库类型读取呀。然后直接匹配二级，相当于顺序换一下，这个文件同时规定了二级分类和二级分类对应的目录，执行时候生成一个每个文件夹自己category.yml。"
这段我没太看懂。。。
tmdb的分类是在请求tmdb名字的时候从tmdb那拿到的数据。这个时候确实是拿来匹配二级分类，然后根据目录里的媒体目录的具体位置放进去的。看起来你似乎让这个过程更加复杂化？
